### ISSUE TITLE
globalThis is not defined (in node < v12)

### ISSUE THREAD

**Issue by rdev-software** [stephenh/ts-proto#97]
After using optionals, ie by importing Google's wrapped types:

```
import "google/protobuf/wrappers.proto";
google.protobuf.StringValue email = 1;
```

TS-Proto plugin adds additional ts files in Google folder. This contains file: _wrappers.ts_

This file uses _globalThis_ which is only available in node 12+ 

Could you please suggest right way of using it in lower node versions ? 
For now I included https://www.npmjs.com/package/globalthis polyfil and by force added
import statement to top of generated wrappers.ts file:
`var globalThis = require('globalthis')();`

**Comment by stephenh** [comment]
Hm, yeah, what you did seems like a good approach. I'd rather not add `globalthis` as a runtime dependency and always include it.

I'd be fine with a config option like `--ts_proto_opt=requireGlobalThisPolyfill=true` that would then put that `var globalThis ...` line at the top of the file.

And then add it to the docs that node-v10 should set that flag.

Does that seem like a good option?

**Comment by webmaster128** [comment]
According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis this can easily be written without a dependency:

```js
var globalThis = globalThis || Function('return this')();
```

**Comment by stephenh** [comment]
@webmaster128 it looks like that won't work in some Content-Security-Policies (mentioned on the page you linked to). That snippet is also not valid TS due to variable-used-before-declared (I get its valid JS, but the ts-proto output is TS).

I think this would need something more like (also from that page):

```
var getGlobal = function () {
  if (typeof self !== 'undefined') { return self; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  throw new Error('unable to locate global object');
};

var globals = getGlobal();
```

Which is both CSP-compliant + would be valid TS.

**Comment by webmaster128** [comment]
@stephenh shounds good. You probably want to try using `globalThis` first, which is available in all modern environments. In any case, this should be easy to include here without adding a dependency.

**Comment by webmaster128** [comment]
If you would be happy with this approach, we would work on a PR working out the details.

**Comment by stephenh** [comment]
@webmaster128 yep, that'd be great. When this was going to be done via a `require`-d polyfill, I wanted it to be conditional based on a flag, but if we're using an inlined function, then I'm fine with going with no flag and just always having the `const globalThis = findGlobalThis();` type approach in the output.

**Comment by stephenh** [comment]
Fixed in v1.43.0. Thanks @willclarktech!
